src/firstboot/firstboot.c

```diff
@@ -285,6 +288,83 @@ static int process_locale(void) {
         return 0;
 }
 
+static int prompt_keymap(void) {
+        _cleanup_strv_free_ char **kmaps = NULL;
+        int r;
+
+        if (arg_keymap)
+                return 0;
+
+        if (!arg_prompt_keymap)
+                return 0;
+
+        r = get_keymaps(&kmaps);
+        if (r < 0)
```

Comment:
we should print an error message somewhere about failures that aren't ENOENT. Given that this function logs all other errors, we should hence have code like this here:

```c
…
r = get_keymaps(&kmaps);
if (r == -ENOENT) /* no keymaps installed */
        return r;
if (r < 0)
        return log_error_errno(r, "Failed to read keymaps: %m");
…